v3/tutorials/02-proof-of-existence/index.mdx

```diff
@@ -106,74 +106,65 @@ Therefore, the first step is to remove some files and content from the files in
    tests.rs
    ```
 
-1. Open the `lib.rs` file in a text editor.
+1. Open the `pallets/template/src/lib.rs` file in a text editor.
 
    This file contains code that you can use as a template for a new pallet.
    You won't be using the template code in this tutorial.
    However, you can review the template code to see what it provides before you delete it.
 
-1. Delete all of the lines in the `lib.rs` file.
-
-1. Add the macro required to build both the native Rust binary (`std`) and the WebAssembly (`no_std`) binary.
+1. Replace all contents of `pallets/template/src/lib.rs` with the following skeleton code that includes all of the [***mandatory*** macros for FRAME V2](https://docs.substrate.io/rustdocs/latest/frame_support/attr.pallet.html):
 
    ```rust
+   //! Substrate Proof-of-Existence Pallet
    #![cfg_attr(not(feature = "std"), no_std)]
-   ```
-
-   All of the pallets used in a runtime must be set to compile with the `no_std` features.
-
-1. Add a skeleton set of pallet dependencies and [macros](/v3/runtime/macros) that the custom pallet requires by copying the following code:
 
-   ```rust
    // Re-export pallet items so that they can be accessed from the crate namespace.
    pub use pallet::*;
 
    #[frame_support::pallet]
    pub mod pallet {
-   	use frame_support::pallet_prelude::*;
+   	use frame_support::{pallet_prelude::*, storage::bounded_vec::BoundedVec};
    	use frame_system::pallet_prelude::*;
-   	use sp_std::vec::Vec; // Step 3.1 will include this in `Cargo.toml`
 
-   	#[pallet::config]  // <-- Step 2. code block will replace this.
-   	#[pallet::event]   // <-- Step 3. code block will replace this.
-   	#[pallet::error]   // <-- Step 4. code block will replace this.
+   	// Define the pallet struct placeholder, various pallet function are implemented on it.
    	#[pallet::pallet]
    	#[pallet::generate_store(pub(super) trait Store)]
    	pub struct Pallet<T>(_);
 
-   	#[pallet::storage] // <-- Step 5. code block will replace this.
-   	#[pallet::hooks]
-   	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {}
-   	#[pallet::call]   // <-- Step 6. code block will replace this.
+   	#[pallet::config]  // TODO: Replace me!
+
+   	#[pallet::event]   // TODO: Replace me!
+
+   	#[pallet::error]   // TODO: Replace me!
+
```

Comment:
I think with just a LITTLE more code, this could compile, and then I think it is worth it.

Reply:
You can see the minimal here: https://sacha-l.github.io/collectables-ethdenver-workshop/#/setup

Reply:
We _could_ omit this completely or just put it as comments to have it "compile" to nothing... it's just to lay out the boiler plate after all.

Reply:
i think just including the minimal empty project which compiles above is best

Reply:
OK - made some tweaks and the skeleton builds in the note-template context (need `Event` defined)

Reply:
> You can see the minimal here: https://sacha-l.github.io/collectables-ethdenver-workshop/#/setup
missed this comment before, if we must, I can move to this instead.